pgmpy/base/DAG.py

```diff
@@ -1257,6 +1257,107 @@ def copy(self):
+          of using correlation (as is the case with partial correlation).
+        - Measures the linear relationship between the residuals.
+        - Works for any mixture of categorical and continuous variables.
+        - The value is bounded between 0 and 1:
+        - Strength close to 1 → strong dependence.
+        - Strength close to 0 → conditional independence.
+
+        Parameters
+        ----------
+        data : pandas.DataFrame
+            Dataset to compute edge strengths on.
+
+        edges : tuple, list, or None (default: None)
+            - None: Compute for all DAG edges.
+            - Tuple (X, Y): Compute for edge X → Y.
+            - List of tuples: Compute for selected edges.
+
+        Returns
+        -------
+        dict
+            Dictionary mapping edges to their strength values.
+
+        Examples
+        --------
+        >>> import pandas as pd
+        >>> from pgmpy.base import DAG
+        >>> # Example DAG with edges and data
+        >>> dag = DAG([('X', 'Y'), ('Z', 'Y')])
+        >>> data = pd.DataFrame({'X': [...], 'Y': [...], 'Z': [...]})
+        >>> dag.edge_strength(data)
+        {('X', 'Y'): 0.99, ('Z', 'Y'): 0.98}
+
+        Notes
+        -----
+        - Based on the `ci_pillai` test using XGBoost-based residualization.
+        - Effect size is computed via Pillai's Trace on residuals' canonical correlations.
+
+        """
+
+        from pgmpy.estimators.CITests import pillai_trace
+
+        # If edges is None, compute for all edges in the DAG
+        if edges is None:
+            edges_to_compute = list(self.edges())
+        # If edges is a single edge tuple
+        elif isinstance(edges, tuple) and len(edges) == 2:
+            edges_to_compute = [edges]
+        # If edges is a list of edge tuples
+        else:
+            edges_to_compute = edges
+
+        strengths = {}
+
+        for edge in edges_to_compute:
+            x, y = edge
+
+            # Get parents of x and y using get_parents instead of predecessors
+            pa_X = self.get_parents(x)
+            pa_Y = self.get_parents(y)
+
+            # Check if either x or y is a latent node
+            if (
+                x in self.latents
+                or y in self.latents
+                or any(parent in self.latents for parent in pa_X + pa_Y)
+            ):
+                raise ValueError(
+                    f"Edge {edge} or its parents involve latent variables. Use CausalInference class for "
+                    "advanced causal effect estimation."
+                )
+
+            # Combine parents for conditioning set (excluding x and y themselves)
+            conditioning_set = set(pa_Y) - {x, y}
+
+            # Run CI test and get effect size
+            result = pillai_trace(
+                X=x, Y=y, Z=list(conditioning_set), data=data, boolean=False
+            )
+
+            # Store the edge strength
+            strengths[edge] = result[0]
```

Comment:
I think we should also store the computed strengths in the graph itself. This could help us with plotting functionality. For each edge, we can do something like:
```
self.edges[edge]['strength'] = compute_value
```